### PR TITLE
Update XVIZ parsing to accept non-enveloped messages

### PR DESCRIPTION
**Pull request by marionleborgne**

### CODE REVIEW

File: modules/io/src/common/xviz-data.js [hunk 1]
```diff
@@ -35,15 +35,16 @@ import {XVIZ_FORMAT} from './constants';
 // - arraybuffer which is a JSON string
 // - JSON object
 // - arraybuffer which is a GLB
+// opts.messageType is the XVIZ Envelope 'type', i.e. one of ('xviz/state_update', 'xviz/metadata', etc.)
 export class XVIZData {
-  constructor(data) {
+  constructor(data, messageType) {
```
Comment (twojtasz) [inline modules/io/src/common/xviz-data.js:40]: I'm wondering if we should put this in an object?

`constructor(data, options) {`

Comment (marionleborgne) [reply modules/io/src/common/xviz-data.js]: updated

File: modules/io/src/common/xviz-data.js [hunk 2]
```diff
@@ -35,15 +35,17 @@ import {XVIZ_FORMAT} from './constants';
 // - arraybuffer which is a JSON string
 // - JSON object
 // - arraybuffer which is a GLB
+// opts.messageType is the XVIZ Envelope 'type', i.e. one of ('xviz/state_update', 'xviz/metadata', etc.)
```
Comment (twojtasz) [inline modules/io/src/common/xviz-data.js:38]: This is a bit confusing actually.  I think we want
`opts.messageType is the Message contained in data.  If supplied it assume data does not have an Envelope.`

File: modules/parser/src/parsers/parse-xviz-message-sync.js [hunk 3]
```diff
@@ -28,7 +28,9 @@ import parseTimesliceDataV2 from './parse-timeslice-data-v2';
 import {getXVIZConfig} from '../config/xviz-config';
 
 // Post processes a stream message to make it easy to use for JavaScript applications
+// opts.messageType is the XVIZ Envelope 'type', i.e. one of ('xviz/state_update', 'xviz/metadata', etc.)
```
Comment (twojtasz) [inline modules/parser/src/parsers/parse-xviz-message-sync.js:31]: ditto about XVIZ Message Type and if supplied there is no Envelope

File: modules/parser/src/parsers/parse-xviz-message-sync.js [hunk 4]
```diff
@@ -37,15 +39,17 @@ export function parseXVIZMessageSync(message, onResult, onError, opts) {
...
-    const xvizData = new XVIZData(message);
+    const xvizData = new XVIZData(message, {messageType});
     const xvizMsg = xvizData.message();
 
     // Non-xviz messages will return null
     if (xvizMsg) {
       const data = xvizMsg.data;
 
-      const v2Type = xvizMsg.type || undefined;
+      const v2Type = messageType || xvizMsg.type || undefined;
```
Comment (twojtasz) [inline modules/parser/src/parsers/parse-xviz-message-sync.js:52]: If we are passing XVIZData messageType, wouldnt the result from `xvizData.message()` reflect that, such that we don't need to change this line

File: modules/parser/src/parsers/parse-xviz-message-sync.js [hunk 5]
```diff
@@ -37,15 +39,17 @@ export function parseXVIZMessageSync(message, onResult, onError, opts) {
     return;
   }
 
+  const {messageType} = opts;
+
   try {
-    const xvizData = new XVIZData(message);
+    const xvizData = new XVIZData(message, {messageType});
```
Comment (twojtasz) [inline modules/parser/src/parsers/parse-xviz-message-sync.js:45]: I am thinking we could just pass `opts` through there.  It's a little broad, but what means is we need to be carefull with XVIZData() and parseXVIZData(). as we would be passing opts through both.

Alternatively , we could scope it in the opts, like `opts = {dataOptions = {}, parseOptions = {}}`

File: modules/io/src/common/xviz-data.js (hunk 2 above)
Comment (marionleborgne) [reply modules/io/src/common/xviz-data.js]: you mean `MessageType` no `Message` right?

File: modules/parser/src/parsers/parse-xviz-message-sync.js (hunk 5 above)
Comment (marionleborgne) [reply modules/parser/src/parsers/parse-xviz-message-sync.js]: updated

File: modules/io/src/common/xviz-data.js (hunk 2 above)
Comment (twojtasz) [reply modules/io/src/common/xviz-data.js]: Yes.  so `messageType` is the optional argument that tells us the **type** of the **data**.   lots of similar words for slightly different semantic meaning here so I leave it to your judgement.  but i'd avoid mixing `messageType` and `MessageType`  together ;)

**Pull request merged**